### PR TITLE
Added CI workflow to publish CVAT images

### LINKED ISSUE #7: Host Container on Docker Hub
Could you please connect this repository to docker hub. This way it would be possible just to download to already built container, since the build process is rather lengthy.

### PR DESCRIPTION
**Pull request by azhavoro**
### Motivation and context
resolve #7 

### How has this been tested?
Manually

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [x] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
~~- [ ] I have added tests to cover my changes~~
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
~~- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#
+# Copyright (C) 2018-2021 Intel Corporation
```
Comment (nmanovic) [inline docker-compose.dev.yml:2]:
```suggestion
# Copyright (C) 2021 Intel Corporation
```

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#
+# SPDX-License-Identifier: MIT
+#
+version: '3.3'
+
+services:
+  cvat:
```
Comment (nmanovic) [inline docker-compose.dev.yml:9]: Why do we need the separate file?

File: README.md [hunk 3]
```diff
@@ -103,6 +110,15 @@ Swagger documentation is visiable on allowed hostes, Update environement variabl
 
 Code released under the [MIT License](https://opensource.org/licenses/MIT).
 
+This software uses libraries from the [FFmpeg](https://www.ffmpeg.org) project.
```
Comment (nmanovic) [inline README.md:113]: Probably need to mention that we use only non-GPL code from ffmpeg. Also mention that exact steps to compile the project can be found in dockerfile...

File: LICENSE [hunk 4]
```diff
@@ -20,3 +20,11 @@ OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
 ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
 OR OTHER DEALINGS IN THE SOFTWARE.
  
+This software uses libraries from the [FFmpeg](https://www.ffmpeg.org) project.
```
Comment (nmanovic) [inline LICENSE:23]: Again, I prefer to mention that we use only non-GPL code from the software.

File: Dockerfile [hunk 5]
```diff
@@ -34,10 +34,11 @@ RUN curl -sL https://github.com/cisco/openh264/archive/v${OPENH264_VERSION}.tar.
     make -j5 && make install PREFIX=${PREFIX} && make clean
 
 WORKDIR /tmp/ffmpeg
-RUN curl -sLO https://ffmpeg.org/releases/ffmpeg-${FFMPEG_VERSION}.tar.bz2 && \
-    tar -jx --strip-components=1 -f ffmpeg-${FFMPEG_VERSION}.tar.bz2 && \
+RUN curl -sL https://ffmpeg.org/releases/ffmpeg-${FFMPEG_VERSION}.tar.bz2 --output - | \
+    tar -jx --strip-components=1 && \
     ./configure --disable-nonfree --disable-gpl --enable-libopenh264 --enable-shared --disable-static --prefix="${PREFIX}" && \
-    make -j5 && make install && make distclean
+    make -j5 && make install && make clean && \
```
Comment (nmanovic) [inline Dockerfile:40]: Please add a comment why clean instead of distclean

**Comment by nmanovic** [comment]
@azhavoro , don't forget to add news into CHANGELOG.md

**Review by nmanovic — CHANGES_REQUESTED** [review]
Let's change comments for ffmpeg code and add info to CHANGELOG. Great job!

**Comment by nmanovic** [comment]
@azhavoro , also need to fix codacy issues.

**Comment by nmanovic** [comment]
@azhavoro , any updates?

**Review by nmanovic — APPROVED** [review]
Great job! I didn't test it, just reviewed the code. For me it looks fine.

**Comment by nmanovic** [comment]
@azhavoro , could you please notify all stakeholders about the achievement (gitter, Intel teams)? Need to clear emphasize why it is important (speed of the build, easy to deploy, alternative channel to distribute CVAT, etc).

**Comment by azhavoro** [comment]
> @azhavoro , could you please notify all stakeholders about the achievement (gitter, Intel teams)? Need to clear emphasize why it is important (speed of the build, easy to deploy, alternative channel to distribute CVAT, etc).

Sure

**Pull request merged**